reposerver/repository/repository.proto

```diff
@@ -141,6 +142,34 @@ message KustomizeAppSpec {
 message DirectoryAppSpec {
 }
 
+message ParameterAnnouncement {
+    // name is the name identifying a parameter.
+    string name = 1;
+    // title is a human-readable text of the parameter name.
+    string title = 2;
+    // tooltip is a human-readable description of the parameter.
+    string tooltip = 3;
+    // required defines if this given parameter is mandatory.
+    bool required = 4;
+    // itemType determines the primitive data type represented by the parameter. Parameters are always encoded as
+    // strings, but this field lets them be interpreted as other primitive types.
+    string itemType = 5;
+    // collectionType is the type of value this parameter holds - either a single value (a string) or a collection
+    // (array or map). If collectionType is set, only the field with that type will be used. If collectionType is not
+    // set, `string` is the default. If collectionType is set to an invalid value, a validation error is thrown.
+    string collectionType = 6;
```

Comment:
why not use `oneOf`?

https://developers.google.com/protocol-buffers/docs/proto3#oneof

Reply:
At first I thought oneOf was like an enum, which would be pretty handy. But looks like it's for when you have multiple fields, only one of which should be set. There are a couple ways I could imagine using this:

```protobuf
oneof collectionType {
    bool isString = 6;
    bool isArray = 7;
    bool isMap = 8;
}
```

Feels a bit unwieldy. I'm not sure it's better than a string.

The other way is this, which would be really nice:

```protobuf
oneof value {
    string string = 6;
    repeated string array = 7;
    map<string, string> map = 8;
}
```

Unfortunately oneOf doesn't support repeated or maps.

Did you have something else in mind?

Reply:
The second way was what I was thinking, but I didn't realize oneOf doesn't support repeated or maps. Carry on!